IOTATransferRestRequestTemplate.py

```diff
@@ -127,19 +124,25 @@ def create_transaction_dictionary(depth=100, request_tag=None):
 
 
 def requestData():
+    # type: () -> Optional[Text]
     request_id = register_request().bytes
     transaction = create_transaction_dictionary(request_tag=request_id)
-    iota.send_transfer(
+
+    # :todo: Populate these values and/or make API object globally-accessible.
+    api = create_iota_object(uri=uri, auth_token=auth_token, seed=seed)
+    api.send_transfer(
         depth=transaction['depth'],
 
         # One or more :py:class:`ProposedTransaction` objects to add to the
         # bundle.
-        transfers=[transaction],
+        transfers=[transaction['transaction-object']],
     )
-    request_dict = create_request(headers={"auth_token": transaction.branch_transaction_hash})
+    request_dict = create_request(headers={'auth_token': transaction['transaction-object'].bundle_hash})
```

Comment:
One more thing.  Sorry, totally spaced on this.

When attaching a transaction to the Tangle, you must "approve" two other transactions, known as the "trunk" and "branch" transactions, respectively.

- The "trunk" transaction is often used to link transactions in the same bundle together, and it may carry additional meaning for certain meta-protocols built on top of IOTA.
- The "branch" transaction is generally arbitrary, although like the trunk transaction, it may be significant for other protocols built on top of IOTA.

However, neither of these values are useful for identifying a transaction, as over time the Tangle will accumulate many transactions with the same trunk/branch transactions.

If you want to identify a transaction uniquely, your best bet is either the transaction hash or the bundle hash:

- The transaction hash is the unique identifier of the transaction.  This is useful if you want to pick out one specific transaction.  However, the transaction hash itself might not be enough to _validate_ the transaction.  That's why there's also:
- The bundle hash is a unique identifier that is shared among all transactions in a single bundle.  Bundles are used to group transactions that are related somehow.  For example, if you want to transfer IOTAs to another address, you must include at least 2 transactions in the same bundle.  The first transaction spends the IOTAs, and the second transaction authorizes the spending of the inputs using a cryptographic signature.

Ultimately, I think the bundle hash is the best identifier to use if you want to transmit a value that can be used to load and verify the payment for services.

Note that AFAIK IOTA transactions are not subject to transaction malleability, but I'm no expert on this.